File: Backend/API/src/main/java/com/example/localhistory/quiz/dto/response/QuizSubmissionResultDTO.java
package com.example.localhistory.quiz.dto.response;

import lombok.Data;

import java.util.List;

@Data
public class QuizSubmissionResultDTO {
    private Long quizId;
    private int totalQuestions;
    private int answeredQuestions;
    private int correctAnswers;
    private int awardedPoints;
    private List<QuizQuestionResultDTO> results;
}

File: Frontend/app/src/main/java/com/example/localhistory/model/response/QuizSubmissionResultDTO.kt
package com.example.localhistory.model.response

data class QuizSubmissionResultDTO(
    val quizId: Long,
    val totalQuestions: Int,
    val answeredQuestions: Int,
    val correctAnswers: Int,
    val awardedPoints: Int,
    val alreadyCompleted: Boolean = false,
    val results: List<QuizQuestionResultDTO>
)
